matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/summary/RoomSummaryDataSource.kt

```diff
@@ -197,6 +198,18 @@ internal class RoomSummaryDataSource @Inject constructor(
         }
         val dataSourceFactory = realmDataSourceFactory.map {
             roomSummaryMapper.map(it)
+        }.map { roomSummary ->
+            val parents = roomSummary.flattenParentIds.mapNotNull { parentId ->
+                getRoomSummary(parentId)?.let { parentSummary ->
+                    SpaceParentInfo(
+                            parentId = parentSummary.flattenParentIds.firstOrNull(),
+                            roomSummary = parentSummary,
+                            canonical = true,
+                            viaServers = emptyList()
+                    )
+                }
+            }
+            roomSummary.copy(spaceParents = parents)
```

Comment:
it doesn't feel right to copy flattenParentes into spaceParents (supposed to be the real m.space.parent)

Reply:
If we don't wanna do this or the approach done earlier, do we have any way of knowing a room's space parent? (when there isn't a space parent event occurring)

Reply:
I mean you could do like that but maybe create a new field in the parent model like `flattenParents` or any name that would mean parent or grand parent, etc.. or `childOf`. It's just to reflect that it's the consequence of a `m.space.child`relation and not of a `m.space.parent`. You could then have a list of RoomSummary and not ParentInfo that needs a viaServer/canonical info.

Other solution could be to get the roomSummary from the flattenParentsIds in the view model.